### PR TITLE
Prevent multiple active sessions

### PR DESCRIPTION
**Pull request by richard-dds**
https://www.pivotaltracker.com/story/show/165518844

Only allow one active session for a given user. Achieves this by saving the user's last session id in the database and manually deleting that previous session when the user logs in again.

This feature is disabled in dev for convenience, but can be enabled by setting`LIMIT_CONCURRENT_SESSIONS = true` in the configuration.

### CODE REVIEW

File: alembic/versions/ab1167fc8260_add_user_last_session_id.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# revision identifiers, used by Alembic.
+revision = 'ab1167fc8260'
+down_revision = '404bb5bb3a0e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('users', sa.Column('last_session_id', postgresql.UUID(as_uuid=True), server_default=sa.text('uuid_generate_v4()'), nullable=True))
```
Comment (dandds) [inline alembic/versions/ab1167fc8260_add_user_last_session_id.py]: Do we need a server default on this column? It's always external data, right?

File: atst/utils/session_limiter.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from atst.domain.users import Users
+
+
+class SessionLimiter(object):
+    def __init__(self, config, session, redis):
+        self.limit_logins = config["LIMIT_CONCURRENT_SESSIONS"]
```
Comment (dandds) [inline atst/utils/session_limiter.py]: Would it be better to just pass this value in by itself, rather than the whole config?

Comment (rthoward) [reply atst/utils/session_limiter.py]: I'm not sure. In this case, it does feel a little unnecessary to pass the entire config, but I could imagine that with a more complex object with more config, having a bunch of config params to pass in the constructor might be too cumbersome.

File: atst/app.py [hunk 3]
```diff
@@ -249,10 +254,18 @@ def make_mailer(app):
...
+
+
 def make_notification_sender(app):
     app.notification_sender = NotificationSender(queue)
 
 
+def make_session_limiter(app, session, config):
+    app.session_limiter = SessionLimiter(config, session, app.redis)
+
+
```
Comment (dandds) [inline atst/app.py]: Looks like this function made it in twice.

Comment (rthoward) [reply atst/app.py]: Whoops! Must have been a bad rebase.

Comment (richard-dds) [reply atst/app.py]: Whoops! Must have been a bad rebase

File: atst/utils/session_limiter.py (hunk 2 above)
Comment (richard-dds) [reply atst/utils/session_limiter.py]: I'm not sure. In this case, it does feel a little unnecessary to pass the entire config, but I could imagine that with a more complex object with more config, having a bunch of config params to pass in the constructor might be too cumbersome.

**Pull request merged**